common/clock/event_time_source.go

```diff
@@ -130,13 +130,19 @@ func (t *fakeTimer) Reset(d time.Duration) bool {
 	t.timeSource.mu.Lock()
 	defer t.timeSource.mu.Unlock()
 
-	if t.done {
-		return false
+	if d < 0 {
+		d = 0
 	}
 
+	wasDone := t.done
 	t.deadline = t.timeSource.now.Add(d)
+	if t.done {
+		t.done = false
+		t.index = len(t.timeSource.timers)
+		t.timeSource.timers = append(t.timeSource.timers, t)
+	}
 	t.timeSource.fireTimers()
-	return true
+	return !wasDone
```

Comment:
I had to take some time to understand all the double-negatives here to verify that this is right, but I'm now convinced it is:

```
// For a Timer created with AfterFunc(d, f), Reset either reschedules
// when f will run, in which case Reset returns true, or schedules f
// to run again, in which case it returns false.
```

I think adding a comment as to why this returns `!wasDone` would help future readers.

Reply:
The function comment explains the return value so I'm not sure it needs another. I renamed it and reversed the meaning so it's more clear.